### ISSUE TITLE
Overlap with strongly-typed literal proposal (and maybe pattern matching?)

### ISSUE THREAD

**Issue by dead-claudia** [tc39/proposal-extended-numeric-literals#6]
I noticed some similarities between this (esp. if #3 is allowed) and [the strongly-typed literal proposal](https://github.com/mikewest/tc39-proposal-literals), and thought these could potentially inform each other. In particular, you could use extensible literals instead of `createFromLiteral` for the general case, making that idiomatic and the escape hatch much more glaringly obvious.

Oh, and also, the possible "extensible map/array literals" has even *more* overlap with [the pattern matching proposal](https://github.com/tc39/proposal-pattern-matching), where if you extend that with the obvious extensible destructuring, such extensible map/array literals are *effectively* the opposite of that (and have come up more than once in that proposal's issues).

**Comment by littledan** [comment]
Yes, there's an intersection with the strongly typed literals proposal. We could give the object here an internal slot to brand it as a literal, making this proposal (possibly) more useful. This is something to add to the explainer.

I'm going to leave extensible map/array literals for a future proposal and out of scope for this one.
